### ISSUE TITLE
manual_contains should not fire when both sides of the expression use the slice element

### ISSUE THREAD

**Issue by Arnavion** [rust-lang/rust-clippy#17563]
### Summary

See below.

### Lint Name

manual_contains

### Reproducer

I tried this code:

```rust
#![deny(clippy::manual_contains)]

pub fn foo(s: &[u8], m: u8) -> bool {
    s.iter().any(|&i| i & m == i)
}
```

I saw this happen:

```
error: using `contains()` instead of `iter().any()` is more efficient
 --> src/lib.rs:4:5
  |
4 |     s.iter().any(|&i| i & m == i)
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `s.contains(&(i & m))`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_contains
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![deny(clippy::manual_contains)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^
```

I expected to see this happen: Lint should not fire. Its suggestion is invalid since `i & m` uses the iterator element `i` itself. It should only fire for `<element binding> == <expression that does not use element binding>` and `<expression that does not use element binding> == <element binding>`

### Version

```text
rustc 1.99.0-nightly (969b803cb 2026-08-09)
binary: rustc
commit-hash: 969b803cbe1d4499f841ae0a49c637d8c70a0458
commit-date: 2026-08-09
host: x86_64-unknown-linux-gnu
release: 1.99.0-nightly
LLVM version: 23.1.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
